### PR TITLE
Publish inline sources and Context docs strings

### LINKED ISSUE #194: Docs for withUserKey are incomplete
https://textileio.github.io/js-threads/classes/_textile_context.context.html#withapikey

It says it just takes a parameter `value` or string. But think it's something like,

```
{
key: string,
secret: string,
type: 1 == user, 2 == account ? or 0 == account
}
```

### LINKED ISSUE #223: Publish actual source in packages
Some packagers will give you warnings like, 

```
Could not load source file "../../src/thread/index.ts" in source map of "../node_modules/@textile/threads-core/dist/thread/index.js".
```

We should package the actually TS source with the releases too (or maybe something just needs an update in the package.json if they already are).

### PR DESCRIPTION
**Pull request by carsonfarmer**
Signed-off-by: Carson Farmer <carson.farmer@gmail.com>

## Description

This PR addresses several existing issues. It also filters out our tests from our published npm packages, but adds inline sources to our source maps.

Fixes #223 
Fixes #194

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

These are all administrative changes, so they should not affect existing tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/id/package.json [hunk 1]
```diff
@@ -33,4 +33,4 @@
     "multibase": "^0.7.0",
     "varint": "^5.0.0"
   }
-}
+}
```
Comment (andrewxhill) [inline packages/id/package.json:36]: are these something new about your default settings?

Comment (carsonfarmer) [reply packages/id/package.json:36]: Yes, these sub package.json files entries now filter out *spec* files and their maps when publishing, but still build them locally for local dev.

Comment (andrewxhill) [reply packages/id/package.json:36]: oh sorry, i meant the removal of the trailing line in files

Comment (carsonfarmer) [reply packages/id/package.json:36]: Ah, yeh that makes sense given the comment was on that line! No this is just my local linter I think?

**Pull request merged**